uaclient/event_logger.py

```diff
@@ -145,6 +177,34 @@ def _generate_failed_services(self):
         }
         return list(set.union(self._failed_services, services_with_error))
 
+    def _process_events_services(self):
+        response = {
+            "_schema_version": JSON_SCHEMA_VERSION,
+            "result": "success" if not self._error_events else "failure",
+            "processed_services": sorted(self._processed_services),
+            "failed_services": sorted(self._generate_failed_services()),
+            "errors": self._error_events,
+            "warnings": self._warning_events,
+            "needs_reboot": self._needs_reboot,
+        }
+
+        print(json.dumps(response, sort_keys=True))
+
+    def _process_events_status(self):
```

Comment:
Instead of having different methods to handle that, I think we should have classes to support that. This will make `process_events` cleaner

Reply:
As discussed with the team, this will be refactored in the future